### PR TITLE
Pin `jest-worker` to `27.2.1` in CRA e2e tests

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | 
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

This should hopefully fix the failing e2e Babel 8 test. I also opened https://github.com/facebook/jest/pull/12128 to fix the regression in Jest (cc @SimenB, in case you missed that PR!), and https://github.com/nodejs/node/issues/41103 to propose completely removing the error.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14040"></a>

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
It works: https://app.circleci.com/pipelines/github/babel/babel/8674/workflows/e6c22401-8160-45a1-8bc0-0886841100a2/jobs/50246

**Review by JLHwung — APPROVED** [review]

**Comment by SimenB** [comment]
I did miss it! 😅 Published your fix now 🙂

**Comment by nicolo-ribaudo** [comment]
Thanks!
I know that the Jest repo generates a lot of notifications, so I thought that pinging you here was a better approach 😛

**Comment by SimenB** [comment]
Great plan! 😀

**Review by existentialism — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
I'm merging this until CRA updates their lockfile to use the new `jest-worker` version.

File: scripts/integration-tests/e2e-create-react-app.sh [hunk 1]
```diff
@@ -45,7 +45,23 @@ if [[ "$(node --version)" == v17.* ]]; then
   export NODE_OPTIONS=--openssl-legacy-provider
 fi
 
+# Remove this when https://github.com/facebook/jest/pull/12128 is fixed
+node -e "
+  var pkg = require('./package.json');
+
+  pkg.resolutions = {
+    'jest-worker': '27.3.1'
```
Comment (nicolo-ribaudo) [inline scripts/integration-tests/e2e-create-react-app.sh:53]:
```suggestion
    'jest-worker': '27.4.4'
```

**Comment by nicolo-ribaudo** [comment]
I tried to update jest-worker and it fails, I will investigate 🤔

**Comment by nicolo-ribaudo** [comment]
Uhhh, if I ssh into CircleCI it doesn't fail.

**Comment by nicolo-ribaudo** [comment]
`5348e6a` shows that it's not a problem with `jest-worker`; it's just still using an older version.

**Comment by nicolo-ribaudo** [comment]
@SimenB `jest-worker` is still broken when using the `NODE_OPTIONS` env variable, because of https://github.com/nodejs/node/issues/41103. I'll open a PR with a workaround for Jest, but this really needs to be fixed in Node.js :confused:

**Comment by nicolo-ribaudo** [comment]
I removed the new commits: this PR still pins to `27.3.1`.

**Pull request merged**